tools/sdc-docker-setup.sh

```diff
@@ -332,30 +357,52 @@ shift $((OPTIND - 1))
 #       https://$dc.api.joyent.com
 # - if given without 'https://' prefix: add that automatically
 promptedUser=
+
+# While the URL passed on the command line is not a valid cloudapi URL,
+# prompt for a new one.
 cloudapiUrl=$1
-if [[ -z "$cloudapiUrl" ]]; then
-    defaultCloudapiUrl=https://us-east-1.api.joyent.com
-    #info "Enter the SDC Docker hostname. Press enter for the default."
-    printf "SDC CloudAPI URL [$defaultCloudapiUrl]: "
-    read cloudapiUrl
-    promptedUser=true
-fi
-if [[ -z "$cloudapiUrl" ]]; then
-    portalUrl=https://my.joyent.com
-    cloudapiUrl=$defaultCloudapiUrl
-elif [[ "$cloudapiUrl" == "coal" ]]; then
-    coal=true
-    cloudapiUrl=https://$(ssh -o ConnectTimeout=5 root@10.99.99.7 "vmadm lookup -j alias=cloudapi0 | json -ae 'ext = this.nics.filter(function (nic) { return nic.nic_tag === \"external\"; })[0]; this.ip = ext ? ext.ip : this.nics[0].ip;' ip")
+while true; do
     if [[ -z "$cloudapiUrl" ]]; then
-        fatal "could not find the cloudapi0 zone IP in CoaL"
+        defaultCloudapiUrl=https://us-east-1.api.joyent.com
+        #info "Enter the SDC Docker hostname. Press enter for the default."
+        printf "SDC CloudAPI URL [$defaultCloudapiUrl]: "
+        read cloudapiUrl
+        promptedUser=true
     fi
-elif [[ "${cloudapiUrl/./X}" == "$cloudapiUrl" ]]; then
-    portalUrl=https://my.joyent.com
-    cloudapiUrl=https://$cloudapiUrl.api.joyent.com
-elif [[ "${cloudapiUrl:0:8}" != "https://" ]]; then
-    cloudapiUrl=https://$cloudapiUrl
-fi
-debug "cloudapiUrl: $cloudapiUrl"
+
+    if [[ -z "$cloudapiUrl" ]]; then
+        portalUrl=https://my.joyent.com
+        cloudapiUrl=$defaultCloudapiUrl
+    elif [[ "$cloudapiUrl" == "coal" ]]; then
+        coal=true
+        cloudapiUrl=https://$(ssh -o ConnectTimeout=5 root@10.99.99.7 "vmadm lookup -j alias=cloudapi0 | json -ae 'ext = this.nics.filter(function (nic) { return nic.nic_tag === \"external\"; })[0]; this.ip = ext ? ext.ip : this.nics[0].ip;' ip")
+        if [[ -z "$cloudapiUrl" ]]; then
+            fatal "could not find the cloudapi0 zone IP in CoaL"
+        fi
+    elif [[ "${cloudapiUrl/./X}" == "$cloudapiUrl" ]]; then
+        portalUrl=https://my.joyent.com
+        cloudapiUrl=https://$cloudapiUrl.api.joyent.com
+    elif [[ "${cloudapiUrl:0:8}" != "https://" ]]; then
+        cloudapiUrl=https://$cloudapiUrl
+    fi
+    debug "cloudapiUrl: $cloudapiUrl"
+
+    cloudApiUrlStatus=$(pingCloudAPIUrl "$cloudapiUrl")
+    if [[ "$cloudApiUrlStatus" == "maintenance" ]]; then
+        printf "\"$cloudapiUrl\" seems to be in maintenance, please try another SDC CloudAPI URL\n"
+    elif [[ "$cloudApiUrlStatus" == "unavailable" ]]; then
```

Comment:
I'd drop the "seems" unsure language. Perhaps:  "'$url' is current in maintenance. Please try again later\n"